accepted/0000-pure-mode.md

```diff
@@ -0,0 +1,356 @@
+
+## Configuration
+
+The `mode` can be enabled by a CLI option when running `npm install` or by a setting in `.npmrc`.
+
+## Compatibility
+
+### Different versions of npm
+
+The `pure-mode` will use the same lockfile as `hoisted` mode. Developers can use either mode without changing the lockfile. This makes `pure-mode` backward compatible and easy to implement. The implementation of `pure-mode` will reside in the reification, it will take the ideal-tree and apply it on disk using symlink instead of hoisting.
+
+### Projects not using workspaces
+
+Non-workspace projects _will_ be able to use the `pure-mode` reification. This is useful as a library author, using `pure-mode` gives me the guarantee that my library will work as intended when consumed by projects which use `pure-mode`.
+
+## Prior Art and Alternatives
+
+### [pnpm](https://github.com/pnpm/pnpm)
+
+Package manager with a `pure-mode`.
+This RFC is mostly inspired by pnpm.
+
+### [ied](https://github.com/alexanderGugel/ied)
+
+Package manager with a `pure-mode`.
+Similar to pnpm but unmaintained.
+
+### [nix](https://nixos.wiki/wiki/Nix)
+
+Functional package manager. Can work with npm packages (eg. [node2nix](https://github.com/svanderburg/node2nix) or [nixfromnpm](https://github.com/adnelson/nixfromnpm) )
+
+### [yarn](https://yarnpkg.com/)
+
+Package manager with a `pure-mode` and project manager.
+
+### [Import maps](https://github.com/WICG/import-maps)
+
+Standard supported by [a few browsers](https://caniuse.com/import-maps) and [deno](https://deno.land/) which makes it possible to implement `pure-mode`.
+
+## Unresolved Questions and Bikeshedding
+
+- Should we use symlinks or junctions on Windows? Both of them have drawbacks:
+
+  - Junctions have to be representated by an absolute path, this means that junctions cannot be committed to git or packed into a package.
+  - Symlinks can only be created in elevated shell [or when Windows is in "developer mode"](https://blogs.windows.com/windowsdeveloper/2016/12/02/symlinks-windows-10/#LCiVBWTgQF5s7fmL.97).
+
+  - answer: junctions by default. support for symlinks can be added later as an opt-in if we see value for it.
+
+- How much community code will break when the system forbids access to undeclared dependencies? In other words, how much code needs to be fixed to work properly in `pure-mode`?
+  - Regarding packages with missing dependencies in the package.json files. The vast majority of the package owners are happy to fix them and the rest simply need to know that this fix is necessary to make their package work great with npm. So the number of packages with missing dependencies is expected to drop fast after npm release `pure-mode`. We may want to later add a feature to npm which allows users to locally declare dependencies on behalf of packages, though this should just be a stop-gap solution.
```

Comment:
I've found it's often not productive to assume how maintainers are willing to maintain their packages. While I generally don't disagree, I think it would be helpful to not codify assumptions.

```suggestion
  - Regarding packages with missing dependencies in the `package.json` file. In many cases, package owners are willing to fix such issues. If there are a significant number of issues encountered around this, we can likely expect the of packages with missing dependencies is expected to drop fast after npm release `pure-mode`. We may want to later add a feature to npm which allows users to locally declare dependencies on behalf of packages as a stop-gap, if existing solutions to this are not enough.
```